### PR TITLE
React Hot Module Loading

### PR DESCRIPTION
**Pull request by anaisbetts**
Based on #10, but adds React HMR to the React templates

/cc @MarshallOfSound @malept

### CODE REVIEW

**Comment by anaisbetts** [comment]
I didn't actually test _any_ of these lol, is there an easy way to do it?

File: react/index.js [hunk 1]
```diff
@@ -6,7 +6,9 @@ module.exports = {
     'react@^15.4.1',
     'react-dom@^15.4.1'
   ],
-  devDependencies: [],
+  devDependencies: [
+    'react-hot-loader@^3.0.0-beta.6',
```
Comment (MarshallOfSound) [inline react/index.js]: This has to be a production dependency right?  It then chooses whether to be a noop or actually do things based on `process.env.NODE_ENV === 'production'`

**Comment by MarshallOfSound** [comment]
@paulcbetts Hmmm, the smoke tests make it the template can be generated not really that they work.  😢 

A few manual `npm link`'s and then `forge init --template=react-typescript` might be in order 😢

File: react/index.js (hunk 1 above)
Comment (anaisbetts) [reply react/index.js]: Ya, I just realized this too

**Comment by MarshallOfSound** [comment]
Testing these out now 👍

**Comment by MarshallOfSound** [comment]
Looking good to me, waiting on smoke builds then merging 👍

**Pull request merged**